src/cpp/database/database_queue.hpp

```diff
@@ -355,20 +356,37 @@ class DatabaseEntityQueue : public DatabaseQueue<EntityDiscoveryInfo>
     {
         try
         {
-            auto id = database_->insert(front().second.entity);
-            if (EntityKind::HOST  == front().second.entity->kind ||
-                    EntityKind::USER == front().second.entity->kind ||
-                    EntityKind::PROCESS == front().second.entity->kind ||
-                    EntityKind::LOCATOR == front().second.entity->kind)
+            EntityDiscoveryInfo info = front().second;
+            EntityId id;
+
+            // Insert the entity only if is discovered
+            if (info.discovery_status == details::StatisticsBackendData::DiscoveryStatus::DISCOVERY)
+            {
+                id = database_->insert(info.entity);
+            }
+            else
+            {
+                id = info.entity->id;
+            }
+
+            // Update the entity status
+            database_->change_entity_status(id,
+                    info.discovery_status !=
+                    details::StatisticsBackendData::DiscoveryStatus::UNDISCOVERY);
```

Comment:
The reason why I am calling the database method is because it will change the status of the entity and also check if its references have to change it status. If I only modify here the status of the entity, it refereces will not be updated

Reply:
I add a commentary clarifying it, I would have thought the same if I had reviewed it